cmd/kueuectl/app/delete/delete_workload.go

```diff
@@ -0,0 +1,372 @@
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type GroupVersionResourceName struct {
+	schema.GroupVersionResource
+	Name string
+}
+
+func (g GroupVersionResourceName) String() string {
+	return strings.Join([]string{g.GroupResource().String(), g.Name}, "/")
+}
+
+func GroupVersionResourceWithName(gvr schema.GroupVersionResource, name string) GroupVersionResourceName {
+	return GroupVersionResourceName{GroupVersionResource: gvr, Name: name}
+}
+
+// Run delete a resource
+func (o *WorkloadOptions) Run(ctx context.Context) error {
+	var (
+		workloads               []*v1beta1.Workload
+		haveAssociatedResources bool
+		err                     error
+	)
+
+	if o.DeleteAll {
+		workloads, haveAssociatedResources, err = o.getAllWorkloads(ctx)
+	} else {
+		workloads, haveAssociatedResources, err = o.getWorkloads(ctx)
+	}
+	if err != nil {
+		return err
+	}
+
+	if len(workloads) == 0 {
+		return nil
+	}
+
+	workloadResources, err := o.getWorkloadResources(workloads)
+	if err != nil {
+		return err
+	}
+
+	if o.DryRunStrategy == util.DryRunNone && !o.Confirmed && haveAssociatedResources {
+		var confirmationMessage string
+		if o.DeleteAll {
+			confirmationMessage = "This operation will also delete the jobs associated with these workloads.\n"
+		} else {
+			confirmationMessage = o.generateConfirmationMessage(workloadResources)
+		}
+
+		if !o.confirmation(confirmationMessage) {
+			fmt.Fprintf(o.Out, "Deletion is canceled\n")
+			return nil
+		}
+	}
+
+	if err := o.deleteWorkloads(ctx, workloadResources); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *WorkloadOptions) getAllWorkloads(ctx context.Context) ([]*v1beta1.Workload, bool, error) {
+	var namespace string
+	if !o.AllNamespaces {
+		namespace = o.Namespace
+	}
+
+	list, err := o.Client.Workloads(namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, false, err
+	}
+
+	var haveAssociatedWorkloads bool
+
+	workloads := make([]*v1beta1.Workload, 0, len(list.Items))
+	for index := range list.Items {
+		wl := &list.Items[index]
+		workloads = append(workloads, wl)
+		if len(wl.OwnerReferences) > 0 {
+			haveAssociatedWorkloads = true
+		}
+	}
+
+	return workloads, haveAssociatedWorkloads, nil
+}
+
+func (o *WorkloadOptions) getWorkloads(ctx context.Context) ([]*v1beta1.Workload, bool, error) {
+	var haveAssociatedWorkloads bool
+
+	workloads := make([]*v1beta1.Workload, 0, len(o.Names))
+
+	for _, name := range o.Names {
+		wl, err := o.Client.Workloads(o.Namespace).Get(ctx, name, metav1.GetOptions{})
+		if client.IgnoreNotFound(err) != nil {
+			return nil, false, err
+		}
+		if err != nil {
+			fmt.Fprintln(o.ErrOut, err)
+			continue
+		}
+
+		workloads = append(workloads, wl)
+		if len(wl.OwnerReferences) > 0 {
+			haveAssociatedWorkloads = true
+		}
+	}
+
+	return workloads, haveAssociatedWorkloads, nil
+}
+
+func (o *WorkloadOptions) getWorkloadResources(workloads []*v1beta1.Workload) (map[*v1beta1.Workload][]GroupVersionResourceName, error) {
```

Comment:
Maybe we should not put too much pressure on this now but, the first owner may not be the actual (root) owner of the wl for example in case of  [FluxMiniclusters](https://kueue.sigs.k8s.io/docs/tasks/run/flux_miniclusters/), [deployments](https://github.com/kubernetes-sigs/kueue/tree/main/pkg/controller/jobs/deployment) and probably more, both the fist layer owner and the workload will be recreated.

Reply:
This is a good point! We can address the root owner and delete it. However, a deployment can create multiple workloads. Deleting the deployment will also remove all associated workloads.

Reply:
To ensure we do not lose context, let's continue the discussion at https://github.com/kubernetes-sigs/kueue/pull/3101#discussion_r1768696280.